micro_http/src/common/headers.rs

```diff
@@ -13,39 +17,178 @@ pub enum Header {
     ContentLength,
     /// Header `Content-Type`.
     ContentType,
+    /// Header `Expect`.
+    Expect,
+    /// Header `Transfer-Encoding`.
+    TransferEncoding,
 }
 
 impl Header {
     fn raw(&self) -> &'static [u8] {
         match self {
             Header::ContentLength => b"Content-Length",
             Header::ContentType => b"Content-Type",
+            Header::Expect => b"Expect",
+            Header::TransferEncoding => b"Transfer-Encoding",
+        }
+    }
+
+    fn try_from(string: &[u8]) -> Result<Self, RequestError> {
+        if let Ok(s) = String::from_utf8(string.to_vec()) {
+            match s.trim() {
+                "Content-Length" => Ok(Header::ContentLength),
+                "Content-Type" => Ok(Header::ContentType),
+                "Expect" => Ok(Header::Expect),
+                "Transfer-Encoding" => Ok(Header::TransferEncoding),
+                _ => Err(RequestError::InvalidHeader),
+            }
+        } else {
+            Err(RequestError::InvalidHeader)
         }
     }
 }
 
 /// Wrapper over the list of headers associated with a Request/Response.
 #[derive(Debug)]
 pub struct Headers {
-    headers: HashMap<Header, String>,
+    optional_headers: HashMap<Header, String>,
```

Comment:
How about a different approach here. You just keep all headers in the HashMap and you add accessor functions for the headers you are interested in?

I am not expecting the accessors to be called so often that it will cause a major performance penalty. I find this approach to be cleaner and easier to extend in the future if needed.

```rust
pub struct Headers {
    headers: HashMap<Header, String>,
}

pub content_length(&self) -> i32 {
    match self.headers.get(Header::ContentLength) {
        // This unwrap is safe because you checked that ContentLength has a valid value when you parsed it.
        Some(value) => std::str::FromStr::from_str(value).unwrap(),
        None => 0,
    }
}

.....
```

Reply:
I just noticed that you have accessors below. You would just need to change them a bit.